test/spec/modules/adoceanBidAdapter_spec.js

```diff
@@ -0,0 +1,485 @@
+import { expect } from 'chai';
+import { spec } from 'modules/adoceanBidAdapter.js';
+import { newBidder } from 'src/adapters/bidderFactory.js';
+import { deepClone } from 'src/utils.js';
+
+describe('AdoceanAdapter', function () {
+  const adapter = newBidder(spec);
+
+  describe('inherited functions', function () {
+    it('exists and is a function', function () {
+      expect(adapter.callBids).to.exist.and.to.be.a('function');
+    });
+  });
+
+  describe('isBidRequestValid', function () {
+    const bannerBid = {
+      bidder: 'adocean',
+      params: {
+        masterId: 'tmYF.DMl7ZBq.Nqt2Bq4FutQTJfTpxCOmtNPZoQUDcL.G7',
+        slaveId: 'adoceanmyaozpniqismex',
+        emiter: 'myao.adocean.pl'
+      },
+      adUnitCode: 'adunit-code',
+      mediaTypes: {
+        banner: {
+          sizes: [[300, 250]]
+        }
+      },
+      bidId: '30b31c1838de1e',
+      bidderRequestId: '22edbae2733bf6',
+      auctionId: '1d1a030790a475',
+    };
+
+    it('should return true when required params found', function () {
+      expect(spec.isBidRequestValid(bannerBid)).to.equal(true);
+    });
+
+    it('should return false when required params are not passed', function () {
+      const invalidBid = Object.assign({}, bannerBid);
```

Comment:
można to od razu zrobić w jednym callu: Object.assign({}, bannerBid, {params: {masterId: 0})